### DISCUSSION TITLE
Voicing features

### DISCUSSION THREAD

**Discussion by Bumblebee001** [GrandOrgue/grandorgue#1413] [Developers]
I would like to see voicing features added to GrandOrgue. 

Is there any chance of seeing these any time soon or at all?

Mark

**Comment by oleg68** [comment]
Could you explain what should this feature do?

**Reply by Bumblebee001** [reply]
Are you familiar with HW? This much sought after software allows one to calibrate each note in a selected rank or indeed an entire octave or even an entire rank (all notes at the same time) in-as-far as pitch, volume, tone (treble/bass), wind pressure (subtle variations in tone for longer held notes to mimick what happens in real life and produces a more realistic effect on the ears and brain interpretation). 

It has vertical sliders which can be adjusted up and down for each key for that rank and a side slider which activates all the sliders of each note together for a particular octave and another for the entire rank. 

This is a very important tool as each organ loaded could be tailored to an environment in which it will be played. The acoustic environment is as much a part of the organ as the instrument itself. A dry organ is very dull in sound quality; the reverb and tonal adjustments (that are also essentially done on real pipes when installed and during maintenance and servicing) give the instrument life. Enhancing the treble in a trumpet rank can make the difference between French and German sounding instrument. 

I'm sure there are people who can perhaps define these better than me.

**Comment by oleg68** [comment]
@Bumblebee001 This is already present at the `Organ Setting` dialog. You can do a lot of adjustments of individual pipes, ranks, windchests or of the whole organ there.

**Comment by Bumblebee001** [comment]
Can you clarify please as I seem to be missing it altogether.

**Comment by oleg68** [comment]
1. Run GrandOrgue
2. Load an organ
3. Select `Audio/MIDI` -> `Organ Settings` from the menu
4. You will see a lot of things to adjust

**Comment by Bumblebee001** [comment]
I’ll look into it later as I’m not at home. The features are very likely
not as clear and user friendly as in HW - which is why I missed them
altogether.

**Comment by Bumblebee001** [comment]
Ok I see what you mean. It would be better to represent the settings as
sliders that the way they are. As they are there is no visual cue to see
what you have and how one could adjust things.

Also I always have issues understanding the difference between amplitude
and gain.

There is no treble/ bass enhancer.

There is no dynamic wind setting.

**Reply by oleg68** [reply]
> There is no treble/ bass enhancer.

Do you really need to have this at each pipe level? Or having the single bass/treble balance would be enough for you?

Adjusting treble/bass os the sound requires a lot of computations. If we have this setting individually for each pipe, we have to compute the adjusted sound for each pipe. With a high poliphony levels it may bring very high CPU load.

Instead, processing the whole organ sound (or processing a few audio groups sound) would require dramaticcaly less CPU power.

**Comment by Bumblebee001** [comment]
In jOrgan the concept of flexible wind was created and usually it is a
setting (on/off) on the console of the organ itself rather than the
software

**Comment by Jean-Gabriel1982** [comment]
+1 for voicing feature. 
I'd be a very good "plus" to GO !

**Comment by kerkovits** [comment]
@Bumblebee001 I checked the sources, and it seems that jOrgan does not have a wind model, it just passes some basic flags to fluidsynth. In the source of fluidsynth I found related functions /src/bindings/fluid_cmd.c:3376 -> `fluid_handle_breathmode` and /src/synth/fluid_chan.c:710 `fluid_channel_cc_breath_note_on_off`. From the inline comments and my glimpse on the source code, it seems that this code is tailored for general wind instruments. As the wind supply of an organ is quite different from a human lung (e.g. fluidsynth assumes that new air can come only in the pauses between notes), I doubt that this feature of jOrgan would result in a more realistic sound, it is just an illusionist setting. When implementing such features to GO, we must be cautious to add true realism instead.

If you would like to enhance the GUI, feel free to create a pull request. The best thing in opensource is that if you do not like something, you can customize any part of the code to your taste. If you are not into programming, you may still create an issue for feature request clearly stating that exactly what enhancement would you propose.

**Reply by Bumblebee001** [reply]
As I explained, in jOrgan the "flexible wind", apparently designed by Paul Stratman, is a feature that is included within his sample-sets and does not feature as a function on jOrgan. John Reimer's sample-sets include this feature too; he calls it "wind destabilizer".

We should have the ability to adjust all modalities for (a) each pipe individually (b) for each octaves (groups of 12 pipes; 13 for last octave) AND (c) a master slider that adjusts ALL the other sliders that may have been individually modified.

Voicing "filters" need to allow changes in tone (brightness), frequency (tuning in cents) and amplitude/ gain (volume) individually and NOT applied to the whole organ. 3 set of sliders need to be created (i) individual notes/ pipes of a rank (ii) each octave of that rank (iii) and a master slider for the whole rank.

**Comment by oleg68** [comment]
Tuning and gain is already present for individual pipes. But voicing filters won't be created for individual pipes until the power of computer becomes sufficient for it.

I may suggest to implement voicing filters for audio groups. If you have a sufficient power computer, you may create separate audio groups for octaves or even for individual pipes.

**Reply by Bumblebee001** [reply]
Hauptwerk software already caters for this so I cannot see why it cannot be implimented for GO.

Also, the presentation / GUI needs to be addressed. As it currently is, it is difficult to use.

**Reply by Jean-Gabriel1982** [reply]
I just up this thread because very interested in voicing features on GO like Hauptwerk do. 

For example if I have a mediocre stop that I can't find elsewhere, I'd like to be able to tweak it a little ..

**Reply by organplayer** [reply]
I'm also in favor of wind destabilization in Grandorgue. I would love to be able to adjust treble and bass in settings. I now use other software on the side to adjust organ sound, but it's better per pipe!

**Comment by Bumblebee001** [comment]
I have already requested these features myself. So far the only outcome is
a version of GO with equaliser GO 3.12.1-15.

We are inching closer but I think there’s still more work needed to
simplify matters for the end users.
